presto-main/src/main/java/com/facebook/presto/execution/ThreadWatchdog.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import io.airlift.units.Duration;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * ThreadWatchdog spins up a background task
+ * that interrupts given thread after the timeout expires.
+ */
+public class ThreadWatchdog
+        implements AutoCloseable
+{
+    private final Thread thread;
+    private final ScheduledExecutorService executor;
+    private final Duration timeout;
```

Comment:
This class doesn't need any of these since starting is implicit in construction. I would just remove the start method and inline creating the task.